.github/workflows/setup.yml

```diff
@@ -0,0 +1,29 @@
+name: Development environment setup
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *"
+  push:
+    branches:
+      - dev
+      - rc
```

Comment:
Why would we only test it when it is too late?

Reply:
The question about when and how to best test your CI is something you are better qualified to answer than me.

This PR is more about prompting you to fix a serious lack in your CI that leaves the documented setup instructions having never once worked in 13+ months of attempting to use them.

Reply:
> This PR is more about prompting you to fix a serious lack in your CI that leaves the documented setup instructions having never once worked in 13+ months of attempting to use them.

Serious lack is quite an overstatement. We had 13.5K contributions in the past year (see GitHub's state of the Octoverse), and this is the first time raising this.

Anyways, I don't think the current way this PR/CI is correct, as stated above.

Reply:
I raised this with you personally, pointing out that none of the existing documentation successfully builds a working environment. So have dozens of others.

I'm absolutely, 100% in agreement that this blind toss isn't the correct final approach for how to do this best. But given the lack of documentation such that one of your most senior contributors doesn't know how it works https://github.com/home-assistant/core/pull/84158#issuecomment-1356631889 how could I possibly get it right?

Please take this PR and address the problem. Or provide documentation that tells us all how this is supposed to work. Given that I work in Python and GA all day, every day, it's not a lack of tool/language knowledge that holds me back. It's entirely due to the undocumented, unexplainable complexity and zero documentation that guides the developer on your approach.

I am certain that HA's wide use and wide set of contributors makes this complexity necessary. No disagreement there, but how you want it to work, and how it does work, isn't clear.

Case in point: I'd love to contribute. I have a long list of bugs and features that could be improved. But I've lost nearly 40 hours struggling to get your build/test system working and frankly I never get to those integration bugs/features because your developer setup has never worked any time I've set aside time to work on it.

Reply:
oh, and let's address the real problem here?

> Why would we only test it when it is too late?

Why would you insult and attack contributors, rather than be helpful and guide them in the direction you want? I work on open source every day, and you're the single most insulting and abrasive maintainer I've come across in decades.

Reply:
> Please take this PR and address the problem.

That is not how open source work. If you plan on moving this PR forward yourself, please do, otherwise close it and track the issue instead.

Thanks 👍 

../Frenck

Reply:
I did create an issue. Which you ignored. I created threads in your forum about the issue, which you ignored. And now you are again insulting and abrasive to someone who would happily contribute, happily create PRs to improve the developer docs, if they could just once follow your docs to get a working environment. Which numerous people have told you for years, directly and indirectly.

How Open Source works is that maintains encourage people to contribute. Not hold back docs and leave everyone blind, then insult them for not knowing things they couldn't possibly know.